openid-connect-rp-metadata-choices-1_0.xml

```diff
@@ -305,6 +305,36 @@
       </t>
     </section>
 
+    <section anchor="ImplementationConsiderations" title="Implementation Considerations">
+      <t>
+	To faciliate interoperation with implementations not yet supporting
+	this specification, deployments may choose to include
+	the single-valued metadata parameter alongside
+	the corresponding multi-valued metadata parameter.
+	However, when the Authorization Server understands
+	the multi-valued metadata parameter, it is RECOMMENDED that
+	the multi-valued metadata parameter be used in preference to
+	the single-valued metadata parameter.
```

Comment:
The RS has no way of knowing whether this specification is supported or not through means other than out of band information. I believe the intention from @MichaelFraser1999 was to ensure maximum coverage without the need to know or multiple roundtrips.

```suggestion
	To faciliate interoperability with implementations not supporting
	this specification, deployments SHOULD include
	the single-valued metadata parameter alongside
	the corresponding multi-valued metadata parameter with their
	preferred single value.
```

Reply:
The reason for the SHOULD is that there are parameters that when not provided have a default value that the RS doesn't support. So maybe that is the distinction, SHOULD when the single-valued metadata has a default that the client doesn't support, MAY otherwise?

Reply:
I get the distinction you're making here.  @MichaelFraser1999 also brought up the situation with default values.  I'll think about what to say that takes this distinction into account.